Docs: update fragmentation docs to reflect ipv6

tommyp1ckles opened the pull request:
We support ipv4 as well as ipv6 fragment tracking, so let's update the docs to reflect that.

tommyp1ckles committed:
Docs: update fragmentation docs to reflect ipv6.n

We support ipv4 as well as ipv6 fragment tracking, so
let's update the docs to reflect that.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

qmonnet commented on Documentation/network/concepts/fragmentation.rst:
```diff
@@ -6,7 +6,7 @@
 
-IPv4 Fragment Handling
+Fragment Handling
 ======================
```
Nit:

```suggestion
Fragment Handling
=================
```

qmonnet reviewed (changes requested):
Change looks good, thanks.

Could you please also remove the note at the end of the file? The referenced commit that fixes the bug was added to kernel 5.9, and the minimum requirement for running Cilium is now 5.10.

tommyp1ckles committed:
Docs: remove note about kernel bug.

The referenced fix was added to kernel 5.9 and the min
version cilium now supports is 5.10.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

qmonnet reviewed (approved):
Thank you!
